### ISSUE TITLE
`ssh_key::Certificate` fails to parse certificates created without an expiration date 

### ISSUE THREAD

**Issue by rexxnor** [RustCrypto/SSH#174]
Hi,

Upon trying out the certificate parser to learn about OpenSSH's certificate structure I discovered that certificates generated without an expiry date fail to be parsed with a `Time` error.

When creating a certificate without expiry OpenSSH decides to set the valid_after to 0 and valid_before to i64::Max which is a lot higher than the limit discussed in [ssh-key/src/certificate/unix_time.rs L11-17](https://github.com/RustCrypto/SSH/blob/776d638384759ed00de79bcfe4303720265cb0a7/ssh-key/src/certificate/unix_time.rs#L11-L17). While I do understand that absurd values make no sense I think failing to parse a perfectly valid certificate is bad.

Could you maybe add a special case to UnixTime specifically for `i64::MAX` that denotes the key does not expire?

Steps to reproduce (OpenSSH 9.5):

```bash
$ cd $(mktemp -d)
$ ssh-keygen -f ca

$ ssh-keygen -f test

$ ssh-keygen -I foobar -s ca test.pub
Signed user key test-cert.pub: id "foobar" serial 0 valid forever

$ cargo init --name sshkey_test
$ cargo add ssh_key -F ed25519

$ #cat > src/main.rs << EOF
use std::path::Path;
use ssh_key::certificate::Certificate;

fn main() {
    let mut args = std::env::args();
    args.next().unwrap();
    let cert_file_arg = if let Some(arg) = args.next() {
        arg
    } else {
        eprintln!("Must supply a path to a an OpenSSH certificate file");
        std::process::exit(-1);
    };

    let cert_file_path = Path::new(&cert_file_arg);
    let certificate = Certificate::read_file(cert_file_path);

    dbg!(&certificate);
}
EOF

$ cargo run -- test-cert.pub
   Compiling sshkey_test v0.1.0 (/tmp/tmp.umUnQAJI6F)
    Finished dev [unoptimized + debuginfo] target(s) in 0.55s
     Running `target/debug/sshkey_test test-cert.pub`
[src/main.rs:17] &certificate = Err(
    Time,
)
```

**Comment by tarcieri** [comment]
It would probably just make sense to get rid of `MAX_SECS` if this (mis)feature (ab)uses timestamps this way

**Comment by rexxnor** [comment]
That works as well. I guess the topic should be discussed in the OpenSSH bugtracker if at all because I am not quite sure how I would implement this given the requirement for protocol/format compatibility across versions.

**Comment by tarcieri** [comment]
`MAX_SECS` is now `i64::MAX` as of #175

**Comment by rexxnor** [comment]
I just tried it with the new 0.6.3 release and the sample certificate still fails to parse. 

I think this is due to the max being defined as `i64::MAX` and not `u64::MAX`. According to the [certificate protocol specification](https://github.com/openssh/openssh-portable/blob/master/PROTOCOL.certkeys#L106-L107) valid_before and valid_after should use `uint64` which is `u64` and not [`i64`](https://github.com/RustCrypto/SSH/blob/c05cd479a0fbdd86e3fdfb8574687a7607273d81/ssh-key/src/certificate/unix_time.rs#L12) in Rust.

[This function](https://github.com/openssh/openssh-portable/blob/99a2df5e1994cdcb44ba2187b5f34d0e9190be91/sshkey.c#L2398) that checks certificate time validity in OpenSSH might be of interest.

**Comment by tarcieri** [comment]
Note the value is defined in terms of `u64`: https://github.com/RustCrypto/SSH/pull/175/files#diff-848d070a3feca06736f81c1b69b7b3c5173b241928e64962c65a79898e5edc32R12

However `u64::MAX` isn't a valid Unix timestamp.

I guess I'll need to add a test vector for one of these certificates, but this is absolutely obnoxious, and a unexpiring certificates are a horrible antipattern.

**Comment by tarcieri** [comment]
Fixing this will likely require breaking changes, since I guess we'll need to replace every single place a certificate can expire with an option, since they chose to use a bogus timestamp as a placeholder for unexpiring certificates.

Will go ahead and reopen this, but I'm not looking forward to solving it.

**Comment by chmac** [comment]
@tarcieri Thanks for solving this. I just ran into this problem [using pam-ssh-agent](https://github.com/nresare/pam-ssh-agent/issues/88). I believe this fix is part of the 0.7.0 release. I see there have been several pre and now rc releases of 0.7.0. Do you have any idea when it will land as a stable release?

**Comment by tarcieri** [comment]
@chmac you can follow along on #475
